packages/react/src/components/ContextualMenu/ContextualMenu.base.tsx

```diff
@@ -262,23 +262,31 @@ function usePreviousActiveElement({ hidden, onRestoreFocus }: IContextualMenuPro
         // Make sure that the focus method actually exists
         // In some cases the object might exist but not be a real element.
         // This is primarily for IE 11 and should be removed once IE 11 is no longer in use.
-        previousActiveElement.current?.focus?.();
+
+        previousElementsQueue.current[0].focus?.();
       }
     },
     [onRestoreFocus],
   );
 
   useIsomorphicLayoutEffect(() => {
     if (!hidden) {
-      previousActiveElement.current = targetWindow?.document.activeElement as HTMLElement;
-    } else if (previousActiveElement.current) {
+      const targetElement = targetWindow?.document.activeElement as HTMLElement;
+      const targetPosition = previousElementsQueue.current.indexOf(targetElement);
+
+      if (targetPosition > -1) {
+        previousElementsQueue.current = [...previousElementsQueue.current].slice(targetPosition + 1);
```

Comment:
I don't think you need to create a new array with `previousElementsQueue.current` here, since slice already does a shallow copy

Reply:
yeah, i was trying to remember if slice was destructive or not